src/xmipp/libraries/parallel/mpi_nma_alignment_vol.h

```diff
@@ -0,0 +1,69 @@
+/***************************************************************************
+ *
+ * Authors:  Mohamad Harastani mohamad.harastani@upmc.fr
+ *	         Slavica Jonic slavica.jonic@upmc.fr
+ *           Carlos Oscar Sanchez Sorzano coss.eps@ceu.es
+ *
+ * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+ * Lab. de Bioingenieria, Univ. San Pablo CEU
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.uam.es'
+ ***************************************************************************/
+
+#include <parallel/xmipp_mpi.h>
+#include <reconstruction/nma_alignment_vol.h>
+
+
+/** Class to perform the NMA Alignment for volumes with  MPI parallelization */
+class MpiProgNMAVol: public ProgNmaAlignmentVol
+{
+private:
+    MpiNode *node;
```

Comment:
How about [smart pointers](https://en.cppreference.com/w/cpp/memory/shared_ptr) ?

Reply:
I believe that your idea is to use smart pointers to get rid of the destructor and use rule of zero. I followed the link and tried to keep it simple, but I faced difficulties.
Lets say that all of **node**, **distributor** and **fileMutex** are smart pointers. Then such a command:
"fileMutex = new MpiFileMutex(node)" replaced by "fileMutex = std::make_shared<MpiFileMutex>(node)" doesn't work in principle, because the class MpiFileMutex expects a raw pointer node not a shared pointer. Or at least this is what I could get from the error while compiling.

Reply:
Of course, you would have to change the type of the variables.
To access the raw pointer, you can use get() method

Reply:
Thanks for the hint. All works well now!